algorithms/src/sorting/Kokkos_SortPublicAPI.hpp

```diff
@@ -72,6 +72,66 @@ void sort(const Kokkos::View<DataType, Properties...>& view) {
   exec.fence("Kokkos::sort: fence after sorting");
 }
 
+// ---------------------------------------------------------------
+// overloads supporting a custom comparator
+// ---------------------------------------------------------------
+template <class ExecutionSpace, class ComparatorType, class DataType,
+          class... Properties>
+void sort([[maybe_unused]] const ExecutionSpace& exec,
+          const Kokkos::View<DataType, Properties...>& view,
+          const ComparatorType & comparator)
+{
+  // constraints
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
+  static_assert(
+      ViewType::rank == 1 &&
+          (std::is_same_v<typename ViewType::array_layout, LayoutRight> ||
+           std::is_same_v<typename ViewType::array_layout, LayoutLeft> ||
+	   std::is_same_v<typename ViewType::array_layout, LayoutStride>),
+      "Kokkos::sort with comparator: supports 1D Views with LayoutRight, LayoutLeft or LayoutStride.");
+
+  static_assert(SpaceAccessibility<ExecutionSpace, MemSpace>::accessible,
+                "Kokkos::sort: execution space instance is not able to access "
+                "the memory space of the View argument!");
+
+  if (view.extent(0) <= 1) {
+    return;
+  }
+
+  if constexpr (Impl::better_off_calling_std_sort_v<ExecutionSpace>){
+    auto first = ::Kokkos::Experimental::begin(view);
+    auto last  = ::Kokkos::Experimental::end(view);
+    std::sort(first, last, comparator);
+  } else {
+    Impl::sort_device_view_with_comparator(exec, view, comparator);
+  }
+}
+
+template <class ComparatorType, class DataType, class... Properties>
+void sort(const Kokkos::View<DataType, Properties...>& view,
+          const ComparatorType & comparator)
```

Comment:
I was wondering how that passes the format check but now I see https://github.com/kokkos/kokkos/pull/6253/files#diff-6c86de4f9c79119175ef9917f4d8f5fb896dda5aaa7eec3978c6f2c431c5d640R31

I am going to let that one slide but I don't like that liberal disabling format everywhere.